tools/serverpod_cli/lib/src/generator/dart/client_code_generator.dart

```diff
@@ -19,16 +21,36 @@ class DartClientCodeGenerator extends CodeGenerator {
       serverCode: false,
       config: config,
     );
+    Map<String, String> generatedCode = {};
 
-    return {
-      for (var model in models)
-        if (!model.serverOnly)
-          p.joinAll([
-            ...config.generatedDartClientModelPathParts,
-            ...model.subDirParts,
-            '${model.fileName}.dart',
-          ]): clientSideGenerator.generateModelLibrary(model).generateCode(),
-    };
+    var sealedHierarchies = getSealedHierarchies(models);
+
+    var modelsWithoutSealedHierarchies =
+        getClassesWithoutSealedHierarchies(models);
+
+    for (var sealedHierarchy in sealedHierarchies) {
```

Comment:
Discussion: Radical thought, but instead of actually generating the code here. How about you simple create a list that contains the model and a nullable allocator. Then we can loop through that list at the of this method to generate the code and not have to repeat the `createFilePath` stuff?

Reply:
I think this could help simplify the code a bit here.

Reply:
Same should be applicable on the `server_code_generator`.

Maybe we then don't even need to break out the `createFilePath` method?